### ISSUE TITLE
Wheel Radius adjustment

### ISSUE THREAD

**Issue by Invictaz** [sonik-br/lgff_wheel_adapter#6]
Some games have a minimum setting of wheel radius of 270 degrees. As the modern logitech wheels can have 900 degrees or more, but the spring loaded wheels only have 180 or maybe even lower, how can we adjust or emulate that?

I tried my Logitech Formula GP in Driveclub on PS4, via Gimx and a pc.  The game only allows 270 degrees as minimum for the wheel, while my wheel as a much lower degree of rotation. 

I know the Ras1ution adapter has an ajustable degree in the menu. The Ras1ution 2 even in 20 steps adjustable. 

Is it something that if we have a screen on the Pico or maybe set it over wifi on the Pico W?

**Comment by sonik-br** [comment]
For ffb wheel it's possible (on some devices) to set the desired rotation range.

But I don't see how to emulate more rotation on a device that have a lower physical limited rotation. Can you elaborate?

**Comment by Invictaz** [comment]
If the device has a physical limited we might have to multiply each degree of rotation by 10 and set the in game rotation to 1080 degrees.

**Comment by sonik-br** [comment]
Why it can't be used at all? It's impossible to play?
Can you point me at the location on the project above that does the conversion?

**Comment by Invictaz** [comment]
https://github.com/vsulako/AFFBWheel/blob/82a882662460310e36e8a043420bf319b6afa2db/AFFBWheel/axis.cpp#L11

This does some range conversion

**Comment by sonik-br** [comment]
This happens on Driveclub?

**Comment by Invictaz** [comment]
Yes. The Cronus Zen has a GPC script to increase every wheel turn by +16 because from 0 to 16 the car does not turn at all. 

But on my wheel it might as well be +30 because my wheel can be turned like crazy within Driveclub without doing anything to the car. It is really insensitive and there is no setting to have outer deadzone.

Some people recommend setting the game to hardcore mode and thus having better steering wheel options.

I did find another software to adjust wheel range from 40-900 degrees.

https://github.com/search?q=repo%3Aberarma%2Foversteer%20range&type=code

But I'm not fully clear on what needs to be adjusted. The deadzone, wheel range or something else.

There is no 270 degree mod for these fixed driveshaft wheels. 

And I have to solder your adapter sometime soon to see if it works. If I inject a ps4 controller as Auth device it might work with ps4 also.

**Comment by sonik-br** [comment]
The code above is for FFB wheels that can be forced into a rotation limiter.

Please share the code if you get ps4 auth working. This would help a lot for users with a ps4!

**Comment by Invictaz** [comment]
The code for PS4 Auth is in GP2040-CE. I will try to look it up. It needs a .PEM file for authentication or a Mayflash dongle.

**Comment by sonik-br** [comment]
Ideally it should do auth using an open dualshock4.
Check out the Santroller too. It can do auth by using a common dualshock4.

**Comment by Invictaz** [comment]
If you are closing issues the willingness to investigate further and do research on various outputs the motivation is lowering.

That said 

I will send you the code example.

**Comment by sonik-br** [comment]
You are asking for stuff that have nothing to do with the adapter.

This one (Wheel Radius) is still open. Could be a good feature to be added but I still can't imagine how that would work.
The wheel reports a value let's say it goes from zero to 1024. Zero being fully rotated ccw, 512 being the middle, and 1024 being fully rotated cw.
Your wheel reports this value range. The logitech ps4 wheel also reports this same value range, no matter if it's set to 270 or 900 degrees. The value range is the same. The physical range is not, but I don't see how this would impact the game.

**Comment by Invictaz** [comment]
I don't agree. If I look at the Maxrace F1 adapter which is end of life, yours has the potention to become the open sourced variant If more features are added. If you want to keep the focus narrow I only can help you by finding obscure wheels and dump them but I'm not sure you are willing to integrate them. As I'm in Europe I'm now trying to find the missing wheels at a low price for a week already. I'm sure it can be done. 

As for the wheel radius the max values are not the problem. It is how you reach the max values. And how far you have to turn the weel. On a wheel with 900 degrees you reach the max value after full turn. On a wheel with 180 degrees this should be scaled in lineair way otherwise the car will not move on small wheel turns.  This is the most read critique on truck simulation games.

**Comment by sonik-br** [comment]
The adapter already does linear scale.

Support for more HID devices as input is also not the purpose of this adapter. Something like the hid-remapper would be more appropriate for the task.
But it's a simple work to do if I have the descriptors. Won't take much of my available time so it's doable.

**Comment by Invictaz** [comment]
HID-remapper is a very cool project but doesn't support Xbox 360, Xbox One, Xbox Series, PS3, PSVita and isn't focussed on wheel support or wheel conversion to a normal controller. Their PS4 support is only an arcade stick and I don't know how they do authentication.

**Comment by Invictaz** [comment]
I got the Brook Ras1ution X now so I can try to dump the values @sonik-br if you guide me trough it. 
My G29 unfortunately isn't rebuild yet, waiting for some parts.

**Comment by sonik-br** [comment]
Dump what?

**Comment by Invictaz** [comment]
Dump the behavior of the usb via Wireshare when we adjust the settings on the Ras1ution.

**Comment by Invictaz** [comment]
I might be able to continue this but it all depends on learning code and other wheel to Pico projects doing some grond work.
